src/lib/openjp2/pi.c

```diff
@@ -1237,7 +1237,13 @@ opj_pi_iterator_t *opj_pi_create_decode(opj_image_t *p_image,
 	l_current_pi = l_pi;
 
 	/* memory allocation for include */
-	l_current_pi->include = (OPJ_INT16*) opj_calloc((l_tcp->numlayers +1) * l_step_l, sizeof(OPJ_INT16));
+	/* prevent an integer overflow issue */
+	l_current_pi->include = 00;
+	if (l_step_l && l_tcp->numlayers < ((OPJ_UINT32)-1) / sizeof(OPJ_INT16) / l_step_l - 1)
```

Comment:
`SIZE_MAX` or `((size_t)-1)` would be the real limit, because `obj_calloc` takes `size_t` parameters. On 64 bit platforms `size_t` is larger than `OBJ_UINT32`.